encoders/nlp/TransformerTFEncoder/__init__.py

```diff
@@ -72,6 +72,8 @@ def __init__(
         self.truncation_strategy = truncation_strategy
         self.model_save_path = model_save_path
 
+        self._padding_strategy = 'max_length' if self.max_length else 'longest'
```

Comment:
I know it is now missing, but could a test be added with a max_length set so that this parameter is tested?

Reply:
@JoanFM  It's good that you mentioned this. If I do this, `test_save_and_load_config` test fails - because nothing gets saved in `save_config()`, so `assert encoder_loaded.max_length == encoder.max_length` fails.


When I say nothing, I mean no attributes specific to `TransformerTorchEncoder`, only params from `BaseExecutor`. The reason that tests were passing so far was that the model was then initiated with default args, and only `max_length` was checked - which was conveniently equal to default `None` in all test examples.

So I think the problem might be that `BaseExecutor` is not saving other arguments at all, or that it offers an interface to save them, but the subclasses are not using it.

So either we investigate this issue first (although I think this is out of scope for this PR) and merge once this is fixed, or we ignore for now, merge, and add the extra test once the issue with saving config is fixed.

Reply:
could you please add a test where the problem appears?

Reply:
done

Reply:
it would be cool also to test that 2 sentences that start the same but differ in some words at the end have the same encoding when using a shorter max_length for padding?

Reply:
@JoanFM nevermind what I said before - the tests were failing locally as I was using version 0.8.3 of jina. Updated and now they pass (as they do on CI).

And sure, I'll add such a test

Reply:
@JoanFM yes, I'll add this test.

Also, tests seem to be passing on CI, but they crash locally. Is it because the version in manifest should be updated on each commit for the tests to run, or is it something else?

Reply:
Is it possible that locally there are some files from previous tests that u did not erase? tests passed properly, no need to raise per each commit

Reply:
I don't know - it's all saved to random directories (tmpdir), so I wouldn't expect this to be a problem. But anyways, CI seems to pass. I have now added the test :)